app/logic/participants.py

```diff
@@ -65,14 +65,25 @@ def userRsvpForEvent(user,  event):
 
     rsvpUser = User.get_by_id(user)
     rsvpEvent = Event.get_by_id(event)
-    program = Program.select(Program).join(ProgramEvent).where(ProgramEvent.event == event).get()
-
-    isEligible = isEligibleForProgram(program, user)
+    if rsvpEvent.singleProgram:
+        program = Program.select(Program).join(ProgramEvent).where(ProgramEvent.event == event).get()
```

Comment:
You don't need to get the program - that's what `rsvpEvent.singleProgram` already is